great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -575,6 +575,11 @@ def get_datasource(
         Returns:
             datasource (Datasource)
         """
+        if datasource_name is None:
+            raise ValueError(
+                "Must provide a datasource_name to retrieve an existing Datasource"
+            )
+
```

Comment:
A bit of additional error checking as parts of the context are liable to call this method with `None` (`_get_data_context_version`)